Hazelcast.Test/Hazelcast.Client.Test/PredicateExtTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Hazelcast.Client.Test.Serialization;
+using Hazelcast.Config;
+using Hazelcast.Core;
+using NUnit.Framework;
+
+namespace Hazelcast.Client.Test
+{
+    [TestFixture]
+    public class PredicateExtTest : SingleMemberBaseTest
+    {
+        private IMap<NamedPortable, NamedPortable> map;
+
+        [SetUp]
+        public void Init()
+        {
+            map = Client.GetMap<NamedPortable, NamedPortable>(TestSupport.RandomString());
+        }
+
+        [TearDown]
+        public void Destroy()
+        {
+            map.Clear();
+        }
+
+        protected override void ConfigureClient(ClientConfig config)
+        {
+            base.ConfigureClient(config);
+            config.GetSerializationConfig().AddPortableFactory(1, new PortableSerializationTest.TestPortableFactory());
+        }
+
+
+        private void FillMap()
+        {
+            for (var i = 0; i < 100; i++)
+            {
+                map.Put(new NamedPortable("key-" + i, i), new NamedPortable("value-" + i, i));
+            }
+        }
+
+
+        [Test]
+        public virtual void TestPredicate_key_property()
+        {
+            FillMap();
+            var predicate = Predicates.Key("myint").GreaterThanOrEqual(50);
+            var values = map.Values(predicate);
+            Assert.AreEqual(50, values.Count);
```

Comment:
This is not map predicate test. Instead it just check the returned predicate also run on map.